src/NLog/Internal/FileAppenders/CountingSingleProcessFileAppender.cs

```diff
@@ -98,7 +111,14 @@ public override void Flush()
                 return;
             }
 
-            this.file.Flush();
+            try
+            {
+                this.file.Flush();
+            }
+            catch (Exception ex)
+            {
+                throw new System.IO.IOException("FileStream Flush Failed: " + FileName, ex);
```

Comment:
I prefer to use the InnerException-pattern instead of modifying the original exception.